runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/AbstractWindmillStream.java

```diff
@@ -130,16 +127,18 @@ private static long debugDuration(long nowMs, long startMs) {
   protected abstract void onNewStream();
 
   /** Returns whether there are any pending requests that should be retried on a stream break. */
-  protected abstract boolean hasPendingRequests();
+  protected boolean hasPendingRequests() {
```

Comment:
revert? Since this is related to correctness a default seems dangerous since it makes it possible to skip implementing correctly.

Reply:
done.